### DISCUSSION TITLE
Annoying popups

### DISCUSSION THREAD

**Discussion by setharnold** [community/community#44621] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hello, in the last week or two, I have noticed that there's vastly more intrusive mouse-over pop-ups throughout much of the github user interface.

For example, on a co-worker's profile page https://github.com/utkarsh2102 if you accidentally mouse-over `@canonical` or `@ubuntu` or any of a dozen other similar texts on this page, there is a very strong chance the pop-up will never go away. Doing a mouse-over on another element can change which pop-up is shown, but the only way to get them to go away entirely is to reload the page.

This is very frustrating. It happens often enough that I thought it was a new design decision, but when typing instructions here it didn't happen reliably. So, this is intermittent. Two others have confirmed this happens with them.

These pop-ups often cover content I want to read. Even when they don't cover content, they are an annoyance.

If this is an intentional design decision, please add a `close` button of some sort to dismiss them. I'm hoping this is an accident, though, and you'll figure out what causes it soon.

My setup is Firefox with noscript (sorry, but it appears all the sites github.com uses are `trusted`).

This feels new in the last week or two.

Thanks

**Comment by corang** [comment]
Same here, I have to keep dodging mouseover popups as I'm trying to use GitHub, firefox no addons that affect github.

**Comment by rgrunber** [comment]
I noticed this as well for a few days but thought it was related to my OS (Fedora Linux). The problem went away for me today without any updates that would have affected this behaviour. Maybe an update has been pushed to certain client-side (Browser) code that is served by GitHub?

**Comment by osma** [comment]
I'm also seeing this starting today, using Firefox 108.0.2 on Ubuntu 22.04. My colleague has the same problem with Firefox on Windows so it's probably not OS specific, but could well be specific to Firefox.

It happens on several kinds of pages when your mouse happens to hover above a user name, triggering a popup that then won't close whatever you do (unless you reload the page). But not every time. For example I just had the problem when commenting on a PR, but I don't see the problem on this page; popups will duly close when you move the mouse away from the user name.

**Reply by rgrunber** [reply]
You're right, it's still happening. I noticed that reloading the page seems to fix the issue for me though.  Looks like it's specific to Firefox (can't reproduce on Chrome, and most of the other issues opened mention Firefox).

![page-popup-bug](https://user-images.githubusercontent.com/1417342/213509008-a7e07d3a-43de-47e3-9616-b934ef7aaf6a.gif)

**Reply by raghvendra-dixit** [reply]
This is repeating again, has anyone noticed as of now ?

**Comment by queenofcorgis** [comment]
Hi there 👋🏼  - I have passed this on to the proper team - they are aware that this is affecting users in the Firefox browser and working to troubleshoot. When I have updates I'll share them with you.

**Reply by SharakPL** [reply]
Please turn it off completely because it's useless (list item content simply duplicated on a freakin popup 🤦‍♂️) and it's like that on all browsers, not just Firefox.

![obraz](https://user-images.githubusercontent.com/5007456/214558710-afedc4f2-8053-491c-a49a-7d6cfe44d470.png)

Update after upgrading to Windows 11:
On Windows 10 these popovers were simply useless. On Windows 11 they are insanely annoying 😭

**Comment by kilograham** [comment]
i have the same issue with firefox on both Ubuntu and macOS... they even manage to nearly always popup and STAY exactly where i want to click next

**Comment by max123kl** [comment]
same here on Debian with FF

**Comment by sbland** [comment]
Same Ubuntu + Firefox. Works ok on Chrome with Windows.

**Comment by zjeffer** [comment]
Same problem on:

* Arch + Firefox 108.0.2. 
* Ubuntu + Firefox 108.0.2

**Comment by dtugend** [comment]
it's horrible for me, I don't even need these pop-ups personally, it gets in the way when hovering while reading and often enough they cover things I want to read and sometimes don't even close
(FireFox 109.0 user on Windows (11))

**Comment by Ghabry** [comment]
This is also terrible for accesibility: I use the addon "Page Zoom WE" in Firefox and have configured:

- Page Zoom 100%
- Text Zoom 120%  (so the text becomes a bit larger, the rest stays the same)

This causes an extremely annoying miscalculation of the popup position: It appears now on-top of the link making it not clickable anymore:

Example in the screenshot: I wanted to click the ``#448 (comment)`` link (see mouse cursor position) but when I hover over it the popup goes in the way and eats the click.

![Screenshot_20230122_143102](https://user-images.githubusercontent.com/1331889/213918437-3e2ee625-5306-4dba-8789-c075452e8b09.png)

**Comment by ar-jan** [comment]
Same problem! Makes working on issues very high friction. I notice that when I keep the Firefox Web Developer Tools open the popups do close again.

**Comment by stuartmclean** [comment]
This keeps popping up when I mouse over my face, blocking the merge button on PRs and then it won't go away unless I refresh the page. 😿 

<img width="949" alt="Screenshot 2023-01-23 at 17 11 19" src="https://user-images.githubusercontent.com/3484465/214090163-78eaa339-6fa0-4cdf-a1b0-75d9e5004403.png">

**Comment by MattTriano** [comment]
I may have found something of a workaround. I was just trying to add github's hovercards to the list of elements uBlock origin blocks, but every time I had a hovercard stuck open, clicking on the uBlock origin extension caused the hovercard to go away.

Still extremely frustrating, and I'd rather not have the hovercards at all, but at least now I have a reliable method to get back to the task that the hovercard interrupted.

**Reply by max123kl** [reply]
Hi @MattTriano ,
your workaround prompted me to check the settings of uBlock Origin.
(It would be interesting to know if everyone who complained here about the "bug" is also using this extension - then it could be causing the error).

When I **block the pop-ups** in uBlock Origin for the domain github.com, FF behaves as expected in all repos.
The pop-up appears on mouse-over and disappears immediately when the mouse is moved further. 
For me, this is an acceptable solution for now - if someone (who knows better how to write English) could inform uBlock Origin about this strange behaviour, I would appreciate it.

**Reply by dtugend** [reply]
I am not using ublock, only plain FireFox, no extensions.
There's a trick to make them go away, but it's still annoying: Click on the stuck pop-up, then click somewhere else on free space.
(I don't even need these pop-ups, they are useless for me.)

**Reply by PartierSP** [reply]
I've been having this same issue too when using Firefox lately.  I don't have uBlock but I am using AdBlock.  I paused it for GitHub and the pop-ups are working as expected again.

But if there was an option to disable these pop ups altogether (like how wikipedia allows their users to decide if page previews are disabled), would be the best solution.

**Reply by akhenry** [reply]
I am not using any extensions at all and I am getting this bug unfortunately.

**Comment by betulum** [comment]
Fixed for me with FireFox 109.0

**Reply by kaspar030** [reply]
Not fixed here with Firefox 109.0, on Arch Linux.

**Reply by SusanRatiLane** [reply]
Not fixed for me, Firefox 109, Intel Mac, Monterey.

**Reply by PartierSP** [reply]
Not fixed FIrefox 109.0+mintlinux+una

**Reply by Ghabry** [reply]
For me this is now fixed but a setting to completely get rid of them would be still useful

**Reply by aldilop** [reply]
Fixed in Firefox 109.0 after refreshing site cache.

**Reply by pgporada** [reply]
Can confirm that in Firefox 109.x if you clear all cache for github.com it'll work as intended.

**Reply by rddunphy** [reply]
Interesting - I thought this had fixed it, but it still happens occasionally. Much less often since clearing the cache, though.

**Comment by rsoika** [comment]
I really wonder why Microsoft floods Github with these unnecessary new features. Github stands out for its simplicity and was a wonderful platform until recently. 

My advice: Please stop it. Github is too important to spoil it with pointless UI gimmicks.

**Reply by jagerman** [reply]
Be careful or they may create and assign you a new Cranky User ~~achievement award~~badge!

**Comment by Phillipus** [comment]
Same problem here.

Windows 11
Firefox 109

**Comment by SharakPL** [comment]
What's the point of these useless popovers on issues and pr lists? It's basically the same information we get on the list item, just doubled on an intrusive popup 🤦‍♂️

And it's like that on all browsers, not just Firefox.
![obraz](https://user-images.githubusercontent.com/5007456/214558710-afedc4f2-8053-491c-a49a-7d6cfe44d470.png)

**Reply by jagerman** [reply]
By making the website more responsive with UI popups contextualizing the flow of high-value information they increase the turn-key availability of the interface and create synergies that have been shown to increase appreciation of Github Enterprise focus-based feedback groups by 3.141592% in 3 different critical assessment categories.

**Reply by SharakPL** [reply]
More like some junior dev at GitHub found out about popovers and decided to poop all over GitHub

**Comment by pneerincx** [comment]
What is taking so long to get this fixed? Just disable any popups all together; they are completely useless and make it impossible to get any work done on the Github website without manually reloading pages :(

**Comment by Phillipus** [comment]
GitHub has become unusable for me now with these popups blocking everything I try to do. Please remove them.

**Comment by berarma** [comment]
Please, get rid of these annoying popups. I have to dodge them while trying to use the site.

**Comment by pdxwolfy** [comment]
Also asking that these annoying popups be disabled. They are not only annoying, but useless to boot since they simply echo information that can already be seen on the page. I did wipe out my cache, and that seems to help with the popups sticking around forever, but the popups are still there reducing usability. If I wanted to use Microsoft Word as my revision control system, I'd get a Windows-based system.

**Reply by joebowbeer** [reply]
> Also asking that these annoying popups be disabled.

Yes, they are still excessive, and not only that, they are giving me PPTSD (post-popup).

**Reply by jagerman** [reply]
Ahhh, now it all makes sense.

![Clippy 2.0](https://www.dashe.com/hubfs/social-suggested-images/Microsoft_Clippy.jpg)

**Comment by programmiri** [comment]
Same pop up behaviour for me:

- macOS Monterey (12.6.1)
- Firefox 109.0 (confirmed in private mode without any addons)

It's a bit hard to use github with FF right now 😅

**Comment by timfenney** [comment]
My experience:

On this page (https://github.com/orgs/community/discussions/44621), the popovers dismiss when mousing away from the popover and target regions. (no bug)

On another page (https://github.com/neovim/neovim/issues/21694), they don't dismiss. However, I can get the desired behaviour by opening the Dev Tools.

Solution: Keep the dev tools open all the time :D

**Comment by Phillipus** [comment]
The workaround I use is to right-click on the popup turd and then left-click somewhere else.

**Reply by SharakPL** [reply]
It doesn't matter if they work properly. Most of them is simply useless and intrusive.

**Comment by rotdrop** [comment]
Please change this feature to be a little bit less intrusive ;) At the very least this new feature does *not* boost the productivity of the Github users.

**Comment by Tanza3D** [comment]
Same issue here. Works fine at 80% zoom and 100% zoom though, but breaks at 90%. Can fix by zooming page out a bit to 80% or in to 100% (Firefox)

**Comment by tort32** [comment]
Same frustration when triaging issues in a list. Seems popup hiding delay was decreased lately but still annoying.
Using Firefox

**Comment by laynor** [comment]
This happens to me as well (firefox 108.0.2, arch linux). Somehow, doesn't happen on this page.

**Comment by efultz-ct** [comment]
when will we get a resolution for this most annoying issue?  Firefox 109.0 (64-bit) - I tried

**Comment by FoamyGuy** [comment]
I've been seeing this issue as well for a few weeks now on Ubuntu with Firefox. 

Hover activated pop-ups that have no apparent way to close them that are covering content on the page. :crossed_fingers: for a fix!

**Comment by riscy** [comment]
If this helps anyone: I have been clicking the popups and pressing 'escape' to make them go away.

**Reply by chris-sloan** [reply]
If the popup is for a user, the click will take you to the user profile. Right click the popup then escape works in this instance.

**Comment by AlCalzone** [comment]
FWIW, this only starts happening for me after I navigate though Github once. It does not happen on the original page load.

As a workaround, I enabled the following TamperMonkey script to hide these popups permanently
```js
// ==UserScript==
// @name         Github: Remove Popover
// @namespace    http://tampermonkey.net/
// @version      0.1
// @match        https://github.com/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @grant        GM_addStyle
// ==/UserScript==

(function() {
    'use strict';

    GM_addStyle('.Popover { display: none !important; }')
})();
```

**Reply by halotroop2288** [reply]
Thank you!

**Comment by ericlingerfelt** [comment]
Same experience for every one on my team as well using Firefox and Chrome on MacOS.

**Comment by jagerman** [comment]
Clearly the problem is that we're trying to use MS infrastructure but not using the "correct" browser (Edge) or OS (Windows).  Oh I can't wait for the resurrection of "made for IE6" badges!  Hey, maybe that can be a Github achievement award!

**Comment by jamieshark** [comment, marked as answer]
Sorry for the disturbance everyone! This was introduced as a part of some accessibility changes we are trying to make throughout the site. We have found a fix for the issue and are working towards resolving it.

**Reply by setharnold** [reply]
Thanks! I haven't noticed this in a while.

**Comment by DhrubajyotiBiswasfounderofDribll** [comment]
Works ok on chrome and edge

**Comment by ericlingerfelt** [comment]
Repaired in Firefox on MacOS 12.6.3.

**Comment by Skydragonsz** [comment]
I don't know if I should put it here or somewhere else but this window is stuck, every page I navigate to on the site has this and it on top of everything else. 
![image](https://github.com/community/community/assets/1357800/47d07f96-7473-49af-86ad-16ad6c932403)
![image](https://github.com/community/community/assets/1357800/8521ecbd-df06-47f3-adc6-544334930181)

**Reply by setharnold** [reply]
Definitely make a new issue; almost no one will see this, here.

**Comment by TheLion** [comment]
> I don't know if I should put it here or somewhere else but this window is stuck, every page I navigate to on the site has this and it on top of everything else. ![image](https://user-images.githubusercontent.com/1357800/246940614-47d07f96-7473-49af-86ad-16ad6c932403.png) ![image](https://user-images.githubusercontent.com/1357800/246940694-8521ecbd-df06-47f3-adc6-544334930181.png)

I have the same issue. Have you created anew issue for this?

**Reply by Skydragonsz** [reply]
I haven't. It didn't show up after a few days anymore so I assumed they fixed it.

**Reply by TheLion** [reply]
Ok. Will see what happens. I noticed this only happens on my iPad, with multiple browsers. Not on my Windows desktop.

**Comment by hadams42** [comment]
Still happening here. Firefox on Windows. Very hard to use the site since this started happening. I have to move the mouse like a video game character and avoid mouse-over "traps" or I cannot do basic things.

**Comment by raghvendra-dixit** [comment]
Seems like this issue has re-surfaced again, i have been noticing since morning, and it's very annoying the patch remains stick on screen and blocks from accessing options beneath it. It also does not goes away using escape button. IF someone from github team can have a look that will be helpful. Thanks

**Comment by TheLion** [comment]
Also still experience this. Hasn't been gone since end of July.

**Comment by FoamyGuy** [comment]
I saw this again today as well. For me it was the first time in a while. Firefox on Linux.

**Comment by JavierJF** [comment]
This is still taking place, specially when you open multiple links in new tabs, and go back to the original tab. The popups persist, and needs to be removed **each time** the tab is revisited.

**Comment by danielmmmm** [comment]
GitHub: We are working on it!

2 years later: EVERYTHING spawns a pop-up and often they don't disappear when moving the mouse away 👿🤬🤬🤬
2 years!!!

**Reply by brc-dd** [reply]
+1, This appears to have regressed. It was fine for some time in between but from 1-2 days started to happen again. Even clicking outside the card doesn't work, and there is weird delay in opening. Looks like they don't cancel some setTimeout on mouseout. You can't even focus on the text box when a popover is open. Someone implemented buggy focus-trap code. 


https://github.com/user-attachments/assets/a6c996f7-5cad-46e0-b938-507a7837fbab

**Comment by ktbarrett** [comment]
It seemed to have been fixed and then I noticed this occurring again, starting yesterday. Windows, Mac, and Linux, on the latest Google Chrome.

**Reply by BitPoet** [reply]
Do you have a blocker active? I've noticed that I only see the pesky popups in Firefox if I enable uBlock Origin. Without it, everything's fine.

**Reply by danielmmmm** [reply]
I use Firefox (v136) and uBlock Origin, but I still get profile pop-ups stuck also when I disable uBlock. On the current page, it takes a while before profiles get stuck. When I reload the page, for the first half or full minute, things work as intended. After a few pop-ups though, some get stuck, and then most get stuck. Other hovercards don't seem to be affected.

Anyways, I just found out that all hovercards can be disabled in the accessibility settings. Hovercards/pop-ups can be useful. But who's idea was it to make everything spawn a hovercard? Like, you have to watch where you park your mouse cursor in order to not get text hidden behind some random hovercard!

**Reply by ktbarrett** [reply]
I don't have a popup blocker. I don't think uBlock origin is even supported on Chrome anymore.

**Comment by RobMayer** [comment]
This is a thing again, it seems.
hover-intent popups won't relinquish focus, and worse yet, when I try to scroll away and navigate elsewhere, the popup steals focus back, and force-scrolls the page back to where the popup is. I'm not one for seemingly hyperbolic claims, but this is really making github hard to use in any meaningful way.

**Comment by dchmelik** [comment]
I have same problem; Slackware GNU/Linux 15+current (up-to-date 64-bit n86) FireFox (stable and maybe ESR) causing downloading hassle: have to open & copy .zip URL (stupidly only one shown) to rename .tar.gz (you should upgrade to .tar.xz), copy date and commit's abbreviated hash number for file-name (Slackware and some other UNIX/GNU/LInux projects tend to name git commit packages software-<ISO_date>_\<hash\>)... on that topic, please upgrade to (international standard) ISO (year-month-day) dates (abbreviated description too vague for programmers, creates many problems... checking software daily, it says today (maybe past midnight) or past week or two and not only you may not recall if you updated, but must fumble around to get exact date for file-name).
